basetest.pm

```diff
@@ -17,6 +17,21 @@ use Mojo::File 'path';
 my $serial_file_pos = 0;
 my $autoinst_log_pos = 0;
 
+my $fail_severity = {
+    '' => 0,
+    'none' => 0,
+    'skipped' => 50,
+    'canceled' => 50,
+    'skipped' => 50,
```

Comment:
You have `skipped` twice. However, I think only ``, `na`, `ok`, `softfail`, `fail` and `canceled` (with only one "l" on os-autoinst level) are relevant here because the other results are only set by openQA (and should not occur on os-autoinst level). But combined with my other suggestion you don't even need `canceled` because supposedly special results like that should always be preserved (as they are supposedly not considered "good" results everywhere else, too). (And technically you could then also drop `fail` because it is the most severe result anyways.)